indico/modules/categories/controllers/display.py

```diff
@@ -167,3 +174,104 @@ def _process(self):
                            Category.chain_titles)
                  .limit(10))
         return jsonify_data(categories=[_serialize_category(c, with_path=True) for c in query], flash=False)
+
+
+class RHDisplayCategory(RHDisplayCategoryBase):
+    def _process(self):
+        past_threshold = self.now - relativedelta(months=1, day=1, hour=0, minute=0, second=0, microsecond=0)
+        future_threshold = self.now + relativedelta(months=1, day=1, hour=0, minute=0, second=0, microsecond=0)
+        event_query = Event.query.with_parent(self.category).order_by(Event.start_dt.desc())
+        past_event_query = event_query.filter(Event.start_dt < past_threshold)
+        future_event_query = event_query.filter(Event.start_dt >= future_threshold)
+        current_event_query = event_query.filter(Event.start_dt >= past_threshold,
+                                                 Event.start_dt < future_threshold)
+        events = current_event_query.filter(Event.start_dt < future_threshold).all()
+        preload_events(events, persons=True)
+        events_by_month = self.group_by_month(events)
+
+        future_event_count = future_event_query.count()
+        past_event_count = past_event_query.count()
+
+        show_past_events = (self.category.id in session.get('fetchPastEventsFrom', set()) or
+                            (session.user and session.user.settings.get('show_past_events')))
+        if show_past_events:
+            past_events = past_event_query.all()
+            preload_events(past_events, persons=True)
+        past_events_by_month = self.group_by_month(past_events) if show_past_events else []
+
+        managers = sorted(self.category.get_manager_list(), key=lambda x: (x.principal_type.name, x.name))
+
+        def threshold_format(dt):
+            return dt.strftime('%Y-%m')
+
+        params = {'event_count': len(events),
+                  'events_by_month': events_by_month,
+                  'format_event_date': self.format_event_date,
+                  'future_event_count': future_event_count,
+                  'future_threshold': threshold_format(future_threshold),
+                  'happening_now': self.happening_now,
+                  'is_recent': self.is_recent,
+                  'managers': managers,
+                  'past_event_count': past_event_count,
+                  'past_events_by_month': past_events_by_month,
+                  'past_threshold': threshold_format(past_threshold)}
+        params.update(get_base_ical_parameters(session.user, self.category, 'category',
+                                               '/export/categ/{0}.ics'.format(self.category.id)))
+
+        if self.category.is_root:
+            if info.HelperMaKaCInfo.getMaKaCInfoInstance().isNewsActive():
```

Comment:
I've moved the news to the new DB; you can use `news = get_recent_news()` here. It'll return an empty list if there are no news or news on the homepage are disabled so no need for extra checks.